sshd-core/src/main/java/org/apache/sshd/common/session/helpers/AbstractSession.java

```diff
@@ -1249,6 +1260,20 @@ protected Buffer encode(Buffer buffer) throws IOException {
         }
     }
 
+    protected void aeadOutgoingBuffer(Buffer buf, int offset, int len) throws Exception {
+        if (outCipher instanceof BaseGCMCipher) {
```

Comment:
I would throw an exception here if this is not a `BaseGCMCipher` since only such ciphers are expected to reach this code